posthog/temporal/batch_exports/batch_exports.py

```diff
@@ -48,6 +48,9 @@
         interval_end={interval_end}
     ) AS persons
 FORMAT ArrowStream
+-- This is half of configured MAX_MEMORY_USAGE for batch exports.
+-- TODO: Make the setting available to all queries.
+SETTINGS max_bytes_before_external_group_by=50000000000
```

Comment:
I've made the setting local to the query just in case it affects other queries in negative ways. Eventually, we can test this out globally for all batch export queries and set it at the client level when connecting. For now, this is the safer choice that still addresses the problem.